tests/simulation/future.cpp

```diff
@@ -0,0 +1,56 @@
+// Copyright 2022 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#include <string>
+#include <thread>
+
+#include "io/future.hpp"
+#include "utils/logging.hpp"
+
+using namespace memgraph::io;
```

Comment:
Not sure how much testing you meant to introduce but I would like to see a small benchmark test for `future` implementation, unless you think that this implementation is trivial enough that it does not require that